### PR TITLE
feat: CF Docker & env changes

### PR DESCRIPTION
**Pull request by iFergal**
Improvements to the Docker deployment and env variable management from @rcmorano - with a traefik compose file also included.

These are migrated from https://github.com/cardano-foundation/keripy/tree/cardano-backer but I've made whatever changes were necessary to get it working again in the new repository.

### CODE REVIEW

File: scripts/start_backer.sh [hunk 1]
```diff
@@ -12,7 +15,7 @@ mkdir -p $CONFIG_DIR/keri/cf
 echo '{
     "backer": {
       "dt": "'$(date -u +"%Y-%m-%dT%H:%M:%S.000000+00:00")'",
-      "curls": ["'$URL':'$PORT'/"]
+      "curls": ["tcp://'$HOST':'$PORT'/", "'$URL':'$PORT'/"]
```
Comment (iFergal) [inline scripts/start_backer.sh]: @rodolfomiranda This works outside of Docker but within the Docker container nothing happens when creating an AID.

Based on `.env.example` - we get `http://cardano-backer:5666` here - is this supposed to be Docket network or external (in my case, localhost)? Neither work for me in Docker. Ports are exposed and mapped directly. (KERIA doesn't complain either)

Comment (iFergal) [reply scripts/start_backer.sh]: Based on Roberto's old deployment - `tcp://cardano-backer:5665` (ending on 5) and `http://localhost:5666` - this works for me, will push some changes (traefik can handle the public domain)

File: scripts/start_backer.sh [hunk 2]
```diff
@@ -9,24 +13,28 @@ else
...
-    "icount": 1,
-    "ncount": 1,
-    "isith": "1",
-    "nsith": "1"
-  }' > $CONFIG_DIR/backer_cfg.json
+cat > $CONFIG_DIR/keri/cf/backer.json <<EOF
+{
+  "backer": {
+    "dt": "$(date -u +"%Y-%m-%dT%H:%M:%S.000000+00:00")",
+    "curls": ["tcp://${INTERNAL_HOST}:$((PORT-1))/", "${URL}:${PORT}"]
```
Comment (iFergal) [inline scripts/start_backer.sh:20]: @rodolfomiranda 2 points here;

Firstly, I only saw `5666` being defined, so presuming `5665` is just port minus 1 - not sure if that's right or it is hard-coded somewhere.

Secondly - in Docker I use `["tcp://cardano-backer:5665/", "http://localhost:5666"]` and this works if KERIA is running outside of Docker as it accesses via localhost. But for local dev I want to deploy KERIA and the backer both in Docker and point KERIA to the backer via the Docker network - in that case, what should this be for that?

`["tcp://cardano-backer:5665/", "http://localhost:5666"]` doesn't work
`["tcp://cardano-backer:5665/", "http://cardano-backer:5666"]` doesn't work

Also tried with `host.docker.internal`

Comment (rodolfomiranda) [reply scripts/start_backer.sh:20]: The tcp port is used for direct mode, so it's not needed unless you also start the backer with the `-T port` argument

The URL used there should be accesible by KERIA, if it's inside a docket, that container should be able to resolve it.

**Pull request merged**